### ISSUE TITLE
Update github action runner

### ISSUE THREAD

**Issue by jlosito** [dependabot/dependabot-core#6584]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

I use GitHub actions extensively as well as dependabot to update packages such as the actions used within workflows. While dependabot does a great job at updating actions, I do not believe it updates the runner versions the workflow is currently using. For example, if I was using the ubuntu-20.04 runner and a new runner was released, ubuntu-22.04, I would like dependabot to notice that and submit a pull request informing me to update it. While it is possible to simply update the runner to used the latest, I would like my builds to be as stable as possible and would like to test the workflow before blindly updating the runner.

**Comment by jeffwidman** [comment]
Agreed, this would be helpful for a lot of folks.

Closing as a duplicate of:
* https://github.com/dependabot/dependabot-core/issues/5541

See also:
* https://github.com/dependabot/dependabot-core/issues/5819
* https://github.com/dependabot/dependabot-core/issues/6247

**Comment by krische** [comment]
@jeffwidman I don't think this should have been closed. This is different than the duplicate since this is looking at the `runs-on` value for jobs in workflows.

**Comment by jeffwidman** [comment]
Good point. I'm no longer on the Dependabot team but perhaps @Nishnha or @jurre could reopen.

**Comment by krische** [comment]
I still think this is a worthwhile feature request and shouldn't be marked stale
